src/server/db_slice.cc

```diff
@@ -1543,7 +1547,9 @@ void DbSlice::OnCbFinish() {
 
 void DbSlice::CallChangeCallbacks(DbIndex id, const ChangeReq& cr) const {
   FetchedItemsRestorer fetched_restorer(&fetched_items_);
+  CounterConditionGuard counter(&counter_flag_);
   for (const auto& ccb : change_cb_) {
+    CHECK(ccb.second);
```

Comment:
sanity, cause it helped me identify the problem. UB is ub, but this one is free and you never know :)